batcher.go

```diff
@@ -250,29 +270,29 @@ func (r *Batcher) Flush() {
 // This tells you how many operations are still in the buffer. This does not include operations that have been sent back to the Watcher as part
 // of a batch for processing.
 func (r *Batcher) OperationsInBuffer() uint32 {
-	return uint32(len(r.buffer))
+	return uint32(len(r.pending)) + uint32(len(r.buffer))
 }
 
 // This tells you how much capacity the Batcher believes it needs to process everything outstanding. Outstanding operations include those in
 // the buffer and operations and any that have been sent as a batch but not marked done yet.
 func (r *Batcher) NeedsCapacity() uint32 {
-	return r.getTarget()
+	return r.Target()
 }
 
-func (r *Batcher) getTarget() uint32 {
+func (r *Batcher) Target() uint32 {
```

Comment:
IMO "TargetCapacity" or even "DesiredCapacity" would be a less vague name for this.

Reply:
There was already a property called "NeedsCapacity" that returns target, so this was just a mistake adding this one. I actually even removed the private property because it wasn't used.